### PR TITLE
Sanitize notificationKey

### PR DESCRIPTION
**Pull request by vforge**
https://wikia-inc.atlassian.net/browse/SUS-543

Sanitize key in WallNotificationController - make XSS impossible.

/cc @mixth-sense @Grunny

### CODE REVIEW

File: extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php [hunk 1]
```diff
@@ -48,10 +48,11 @@ public function Update() {
 		wfProfileIn( __METHOD__ );
 
 		$this->response->setVal( 'alwaysGrouped', empty( $wgEnableWallExt ) && empty( $wgEnableForumExt ) );
-		$this->response->setVal( 'notificationKey', $this->request->getVal( 'notificationKey' ) );
+		$this->response->setVal( 'notificationKey', htmlentities( $this->request->getVal( 'notificationKey' ), ENT_QUOTES ) );
```
Comment (Grunny) [inline extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php]: This is being output into a HTML attribute, so I'd prefer `Sanitizer::encodeAttribute`, and it's also generally preferable to escape as close to the output as possible, so I'd consider escaping in the template as a best practice, but this is OK as well.

**Comment by Grunny** [comment]
One suggestion, but LGTM in general.

**Comment by vforge** [comment]
@Grunny Thanks for the suggestion, I've noticed both approaches in the code - I decided to proceed with your tip.

**Comment by Grunny** [comment]
LGTM.

**Comment by Grunny** [comment]
This also needs to be handled in GlobalNavigationWallNotifications_Update.php.

**Comment by Grunny** [comment]
LGTM and tested.

**Comment by vforge** [comment]
@Grunny Thanks for finishing it up at night!

**Pull request merged**